merge for the release of v1.6.0

wongoo opened the pull request:
**What this PR does**:
- [Imp: add cache in reflection](#179)
- [Imp: decode string perf](#188)
- [Fix: get attachment lost nil key](#191)

zouyx committed:
add license checker (#175)

wongoo committed:
add release note for v1.5.0 (#178)

Co-authored-by: Xin.Zh <dragoncharlie@foxmail.com>

micln committed:
Imp: cache in reflection (#179)

* benchmark result

* use cache in findField

* encode benchmark

* call field() once

* remove version

* fix import sync

* cache in registerPOJO

* add json bench result

* prune unneccessary rtype.Field(index)

* cache comment

* rename cache

* switch to if

* remove return value name

* findFieldWithCache

* remove if check when fieldStruct is nil

Co-authored-by: 望哥 <gelnyang@163.com>

micln committed:
update dependency

micln committed:
rename serialize arg name

htynkn committed:
Create .asf.yaml

zonghaishang committed:
优化hessian解码string性能，提升54%

zonghaishang committed:
optimize code.

zonghaishang committed:
optimize code.

zonghaishang committed:
fix code review.

zonghaishang committed:
optimize codes.

zonghaishang committed:
optimize cods.

zonghaishang committed:
optimize code.

micln committed:
update license

micln committed:
go.sum

micln committed:
Merge remote-tracking branch 'upstream/master' into issue177

micln committed:
ci go version

micln committed:
testify -> 1.4.0

micln committed:
testcase

micln committed:
travis.yml

micln committed:
decode value before reflect find

micln committed:
setvalue

micln committed:
decode nilPtr to nilPtr

fangyincheng committed:
Merge pull request #185 from micln/issue177

Issue177：Upgrade go.mod & LICENSE

champly committed:
fix get attachment lost nil key

champly committed:
manually import package

champly committed:
add ToMapStringString unit test

champly committed:
rename test function name with issue

AlexStocks committed:
Merge pull request #191 from champly/fix-attachment-lost-nil

fix get attachment lost nil key

micln committed:
setmap

zonghaishang committed:
support for decode emoji.

zonghaishang committed:
Merge pull request #188 from zonghaishang/perf_decode_string

Perf decode string

zonghaishang committed:
refactor code

fangyincheng committed:
Merge pull request #194 from zonghaishang/refactor_code

remove unused comments.

AlexStocks commented:
pls wait for the next week. the ut coverage decreases now.

zonghaishang committed:
add unit test.

zonghaishang committed:
add unit tests.

zonghaishang committed:
refactor tests.

AlexStocks committed:
Merge pull request #197 from zonghaishang/more_unit_tests

More unit tests

fangyincheng commented:
decrease 10% in string.go？


![image](https://user-images.githubusercontent.com/18375310/83847697-42941280-a73f-11ea-84a5-7e2a4c617e92.png)

willson-chen committed:
Update travis/main.sh (#200)

- Remove duplicate key 'webhooks'
- Key 'matrix' is an alias for `jobs`, using `jobs`
- Specify the os and dist explicitly

fangyincheng committed:
Mrg: merge master

fangyincheng committed:
Mod: modify

gaoxinge committed:
Code format (#199)

* .gitignore

* code clean

* code clean

* remove length check

fangyincheng committed:
Fix: comments

fangyincheng committed:
Fix: format package

AlexStocks committed:
Merge pull request #201 from fangyincheng/ignore_fields

Ignore fields

willson-chen committed:
Fix #181: float32 accuracy issue (#196)

* Fix #181: float32 accuracy issue

* Fix go fmt failure

* Add the unit test case for Issue181

* Add encFloat32 in double.go to encode float32 type

- Call encFloat32 to encode float32 while encoding
- Add unit test case to test float32 encoding

* Improve encFloat32 of double.go

* Fix git fmt failure

fangyincheng reviewed (approved):
LGTM

wongoo committed:
add release note for v1.6.0 (#202)

* add release note for v1.5.1

* add release note for v1.5.1

* add notice

* update notice

* =fix release note for v1.6.0

wongoo committed:
fix conflict
